### PR TITLE
feat(efb): make callibration deadzone range bars green when within dead range

### PR DESCRIPTION
**Pull request by ErickSharp**
## Summary of Changes

Make the calibration range bars on the efb green when the slider is within their range. Also changes "Range" to "Deadzone +/-"

## Screenshots (if necessary)

image
image

## Additional context

Discord username (if different from GitHub): Erick [Z-6]#6484

## Testing instructions

Go to the EFB, then settings, then at the bottom right click on 'Callibrate', move the throttle and see the slider go into the dead range. 

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: src/instruments/src/EFB/Components/Progress/Progress.tsx [hunk 1]
```diff
@@ -107,39 +107,48 @@ export const ProgressBar: React.FC<ProgressBarProps> = ({
...
+
+    const getBarStyle = () => {
+        if (completedBarBeginValue && completedBarEnd) {
+            const barBegin = parseFloat(completedBarBeginValue);
+            const barEnd = parseFloat((completedBarEnd !== 0 ? (completedBarEnd / 50 - 1) : 0.00).toFixed(2));
+            const roundedThrottlePos = parseFloat(throttlePos.toPrecision(2));
+
+            if (vertical) {
+                if (roundedThrottlePos <= barEnd && roundedThrottlePos >= barBegin) {
+                    return 'absolute z-10 -mt-2.5 h-1.5 bg-green-500';// horizontal progress bar with green bg
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Progress/Progress.tsx:120]:
```suggestion
                    return 'absolute z-10 -mt-2.5 h-1.5 bg-green-500'; // horizontal progress bar with green bg
```

File: src/instruments/src/EFB/Settings/ThrottleConfig/BaseThrottleConfig.tsx [hunk 2]
```diff
@@ -15,30 +15,34 @@ interface Props {
     setInitialize,
 }
 
+export const ThrottleContext = React.createContext<{throttlePos: number}>({ throttlePos: 0 });
```
Comment (Benjozork) [inline src/instruments/src/EFB/Settings/ThrottleConfig/BaseThrottleConfig.tsx:18]: It's one value, let's just get the LVar locally.

**Review by Benjozork — CHANGES_REQUESTED** [review]

File: src/instruments/src/EFB/Settings/ThrottleConfig/BaseThrottleConfig.tsx (hunk 2 above)
Comment (ErickSharp) [reply src/instruments/src/EFB/Settings/ThrottleConfig/BaseThrottleConfig.tsx:18]: you are completely right, i will fix that soon

Comment (ErickSharp) [reply src/instruments/src/EFB/Settings/ThrottleConfig/BaseThrottleConfig.tsx]: I removed the usage of the context, again I am very sorry about that!

File: src/instruments/src/EFB/Components/Progress/Progress.tsx [hunk 3]
```diff
@@ -31,6 +31,7 @@ export const ProgressBar: React.FC<ProgressBarProps> = ({
     completedBarEnd,
     completedBarBegin,
     completedBarBeginValue,
+    rawThrottlePosition,
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Progress/Progress.tsx:34]: This might not only be used for throttles. Let's rename this prop.

**Review by Benjozork — CHANGES_REQUESTED** [review]

File: src/instruments/src/EFB/Components/Progress/Progress.tsx (hunk 3 above)
Comment (ErickSharp) [reply src/instruments/src/EFB/Components/Progress/Progress.tsx:34]: The naming is fine as its the throttle's position

Comment (Benjozork) [reply src/instruments/src/EFB/Components/Progress/Progress.tsx:34]: This component is in a folder that is for generic components.

Comment (ErickSharp) [reply src/instruments/src/EFB/Components/Progress/Progress.tsx]: I forgot about that, thank you. Fixed with latest commit.

File: src/instruments/src/EFB/Components/Progress/Progress.tsx [hunk 4]
```diff
@@ -107,39 +108,46 @@ export const ProgressBar: React.FC<ProgressBarProps> = ({
         alignItems: labelAlignment === 'outside' ? 'center' : 'initial',
     };
 
-    const checkOrientation = () => {
-        if (vertical) {
-            return 'horizontal-progress-bar';
+    const getBarStyle = () => {
+        if (completedBarBeginValue && completedBarEnd && completionValue) {
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Progress/Progress.tsx:112]: **thought:** We might want this feature (green bar) disabled in other progress bars. As this PR adds the feature to the generic component, it should be optional.

**suggestion:** Add a prop to enable this.

File: src/instruments/src/EFB/Components/Progress/Progress.tsx [hunk 5]
```diff
@@ -107,39 +108,46 @@ export const ProgressBar: React.FC<ProgressBarProps> = ({
...
+            const barBegin = parseFloat(completedBarBeginValue);
+            const barEnd = parseFloat((completedBarEnd !== 0 ? (completedBarEnd / 50 - 1) : 0.00).toFixed(2));
+            const roundedThrottlePos = parseFloat(completionValue.toPrecision(2));
```
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Progress/Progress.tsx:115]: **issue:** `roundedThrottlePos` should be renamed to something generic.

**Review by Benjozork — CHANGES_REQUESTED** [review]

File: src/instruments/src/EFB/Components/Progress/Progress.tsx (hunk 4 above)
Comment (ErickSharp) [reply src/instruments/src/EFB/Components/Progress/Progress.tsx:114]: good idea

**Review by Benjozork — APPROVED** [review]

**Comment by davidnorfolk96** [comment]
Quality Assurance Trainee Report

Discord: !davidn123#0001
Object of testing: #5326
Tier of Testing:1
Date: 02/08/2021

Testing Process: Loaded up at LOWI on the runway, Engines one. Loaded up the EFB proceeded to the "calibrate" tab. checked the throttles to see if the slider moves into the green. On IDLE,CLB,FLX,TOGA. All passed & nothing to report in terms of breaks. 

Negatives: N/A

Testing Results: Passed

Conclusions: Followed the testing instructions. No breaks. All worked at intended. 

Media: 
Screenshot_2
Screenshot_4
Screenshot_5

**Comment by NUTTTTTTTT** [comment]
Quality Assurance Trainee Report

Discord: █▀█ █▄█ ▀█▀#2123
Object of testing: #PR 5326
Tier of Testing: 1
Date: 02/08/2021

Testing Process: Spawned on runway at EGCC, fully started. Loaded EFB and opened calibrration page. Checked each detent on calibration page showed as white, changed to green bars when throttle was set into each position. All positions checked and change as expected.

Negatives: N/A

Testing Results: Passed

Conclusions: Worked as expected, no observed difference in perf on runway and calibration page performed as usual.


image

**Pull request merged**